### ISSUE TITLE
Non-exhaustive patterns in function fromString

### ISSUE THREAD

**Issue by vaibhavsagar** [ndmitchell/weeder#2]
Running this on https://github.com/vaibhavsagar/suppandi gives me the following error:

```
weeder: src/Stack.hs:34:9-42: Non-exhaustive patterns in function fromString
```

**Comment by vaibhavsagar** [comment]
Putting a `traceShowId` in the right place gives me this as the input to `map fromString`:

```
[String ".",Object (fromList [("location",Object (fromList [("git",String "https://github.com/vaibhavsagar/duffer.git"),("commit",String "404bad52e35b4641c99abfadd117546879804036")])),("subdirs",Array [String "duffer",String "duffer-json"]),("extra-dep",Bool True)])]
```

I don't understand what the output of `fromString` is meant to be in this case, but I'm happy to help in any way I can :smile:

**Comment by ndmitchell** [comment]
It's choking on the location key in the stack.yaml - I'll fix this shortly.

**Comment by ndmitchell** [comment]
Thanks for the report and the diagnosis work!

**Comment by ndmitchell** [comment]
After fixing that, the results are:

```
*Main> :main C:\Neil\suppandi --build
suppandi-0.1.0.0: Test running disabled by --no-run-tests flag.
= Package suppandi =

== Section test:suppandi-test ==
Redundant build-depends entry
* suppandi

= Package duffer-json =

== Section test:duffer-json-test ==
Module not compiled
* Spec.hs
Redundant build-depends entry
* aeson
* base
* bytestring
* duffer
* duffer-json
* hspec
* process
* utf8-string

= Package duffer =

== Section test:duffer-test ==
Module not compiled
* Spec.hs
Redundant build-depends entry
* QuickCheck
* attoparsec
* base
* byteable
* bytestring
* containers
* digest
* directory
* duffer
* filepath
* hspec
* process
* transformers
* utf8-string
```

I think the suppandi error is genuine because you haven't yet generated the tests. I think the other dependent packages are because Stack isn't bothering to compile their dependencies as they are at a remote location - I think that's just a consequence of tying things up across repos.

**Comment by ndmitchell** [comment]
Released as weeder-0.1.1. Thanks for the report - it lead me to a much more robust and simple way of querying information from stack.yaml
